tests/conftest.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from open_feature import open_feature_api as api
+from open_feature.evaluation_context.evaluation_context import EvaluationContext
+from open_feature.exception.exceptions import OpenFeatureError
+from open_feature.flag_evaluation.error_code import ErrorCode
+from open_feature.flag_evaluation.flag_evaluation_details import FlagEvaluationDetails
+from open_feature.flag_evaluation.flag_type import FlagType
+from open_feature.hooks.hook import Hook
+from open_feature.hooks.hook_context import HookContext
+from open_feature.provider.no_op_provider import NoOpProvider
+
+
+@pytest.fixture()
+def no_op_provider_client():
+    api.set_provider(NoOpProvider())
+    return api.get_client()
+
+
+class TestExceptionHook(Hook):
+    def before(self, hook_context: HookContext, hints: dict) -> EvaluationContext:
+        pass
+
+    def after(
+        self, hook_context: HookContext, details: FlagEvaluationDetails, hints: dict
+    ):
+        raise Exception("Generic exception raised")
+
+    def error(self, hook_context: HookContext, exception: Exception, hints: dict):
+        pass
+
+    def finally_after(self, hook_context: HookContext, hints: dict):
+        pass
+
+    def supports_flag_value_type(self, flag_type: FlagType) -> bool:
+        return True
+
+
+class TestOpenFeatureErrorHook(Hook):
```

Comment:
This is a case where mocking would be more useful- you can also validate that the provider is called with the correct values using mock assertions like

`mock_provider.before.asset_called_with(...)`

Reply:
Another option would be to use a spy, then you can have shared predefined logic with a real class but you would spy on it to run those assertions.

Reply:
The hooks are now mocked